src/KafkaFlow.Admin/Extensions/ClusterConfigurationBuilderExtensions.cs

```diff
@@ -58,5 +67,97 @@ public static class ClusterConfigurationBuilderExtensions
         {
             return cluster.EnableAdminMessages(adminTopic, $"Admin-{Assembly.GetEntryAssembly().GetName().Name}");
         }
+
+        /// <summary>
+        /// Creates the telemetry producer and consumer to send and receive metric messages
+        /// </summary>
+        /// <param name="cluster">The cluster configuration builder</param>
+        /// <param name="topicName">The topic to be used by the metric commands</param>
+        /// <param name="consumerGroup">The consumer group prefix</param>
+        /// <returns></returns>
+        public static IClusterConfigurationBuilder EnableTelemetry(
+            this IClusterConfigurationBuilder cluster,
+            string topicName,
+            string consumerGroup)
+        {
+            cluster.DependencyConfigurator.AddSingleton<IMemoryCache, MemoryCache>();
+            cluster.DependencyConfigurator.AddSingleton<ITelemetryCache, TelemetryCache>();
+
+            var groupId =
+                $"{consumerGroup}-{Environment.MachineName}-{Convert.ToBase64String(Guid.NewGuid().ToByteArray())}";
+
+            var producerName = $"telemetry-{Environment.MachineName}-{Convert.ToBase64String(Guid.NewGuid().ToByteArray())}";
+
+            return cluster
+                .AddProducer(
+                    producerName,
+                    producer => producer
+                        .DefaultTopic(topicName)
+                        .AddMiddlewares(
+                            middlewares => middlewares
+                                .AddSerializer<ProtobufNetSerializer>()))
+                .AddConsumer(
+                    consumer => consumer
+                        .Topic(topicName)
+                        .WithGroupId(groupId)
+                        .WithWorkersCount(1)
+                        .AsReadonly()
+                        .WithBufferSize(10)
+                        .WithAutoOffsetReset(AutoOffsetReset.Latest)
+                        .WithPartitionsAssignedHandler((resolver, partitions) =>
+                        {
+                            TelemetryScheduler.Set(
+                                groupId,
+                                async _ =>
+                                {
+                                    await resolver
+                                        .Resolve<IProducerAccessor>()
+                                        .GetProducer(producerName)
+                                        .BatchProduceAsync(
+                                            resolver
+                                                .Resolve<IConsumerAccessor>()
+                                                .All
+                                                .Where(c => !c.IsReadonly) // TODO GET ONLY CONSUMERS FROM THIS CLUSTER
```

Comment:
How to get the consumers by Cluster? 
Maybe defining a Name/Id in the cluster and exposing it on Consumer interface (ClusterId/Name), so we could filter the consumers here.

Reply:
done

Reply:
Let's create the OnStarted() on cluster and put this logic there

Reply:
done